packages/vite/src/node/utils.ts

```diff
@@ -37,10 +37,26 @@ export const flattenId = (id: string): string =>
 export const normalizeId = (id: string): string =>
   id.replace(/(\s*>\s*)/g, ' > ')
 
+//TODO: revisit later to see if the edge case that "compiling using node v12 code to be run in node v16 in the server" is what we intend to support.
+const builtins = new Set([
+  ...builtinModules,
+  'assert/strict',
+  'diagnostics_channel',
+  'dns/promises',
+  'fs/promises',
+  'path/posix',
+  'path/win32',
+  'readline/promises',
+  'stream/consumers',
+  'stream/promises',
+  'stream/web',
+  'timers/promises',
+  'util/types',
+  'wasi'
+])
+
 export function isBuiltin(id: string): boolean {
-  const deepMatch = id.match(deepImportRE)
-  id = deepMatch ? deepMatch[1] || deepMatch[2] : id
-  return builtins.includes(id)
+  return builtins.has(id.replace(/^node:/, ''))
```

Comment:
I'm a bit late here, but wouldn't a `node:` prefix always mean that the module is built-in? I think that that was the purpose of adding the prefix in the first place.

Thus you could check that first, and then fall back to the list:

```suggestion
  return id.startsWith('node:') || builtins.has(id)
```

Reply:
it's done correctly here.
`node:` is getting replaced to nothing and checked if it exists in node's builtin module list.
someone could have wrote `node:htttp` or something else on accident which would be wrong